### ISSUE TITLE
Adjust Elementor Pro animated-headline speed

### ISSUE THREAD

**Issue by dtbaker** [elementor/elementor#4392]
## Description

It doesn't seem possible to change the Elementor Pro animated text properties. I can see the default in `frontend.js` here:

```
var settings = {
			animationDelay: 2500,
			//letters effect
			lettersDelay: 50,
			//typing effect
			typeLettersDelay: 150,
			selectionDuration: 500,
			//clip effect
			revealDuration: 600,
			revealAnimationDelay: 1500
		};
```

I tried adjusting these default settings by adding various attribute to the element like so, but couldn't get one that seemed to change these default settings.

```
add_action( 'elementor/frontend/widget/before_render', function( $elementor ) {
	if($elementor && $elementor->get_name() == 'animated-headline') {
		  $elementor->add_render_attribute( '_wrapper', 'data-type-letters-delay', 4);
       }
}, 10, 1 );
```
Is there a way to adjust these default settings from Elementor render attributes? If not, it would be a nice enhancement. 

Thanks

**Comment by bainternet** [comment]
Currently, this feature is not supported, but we will look into it in the future if we see a large demand for it.

Thanks for the feedback.

**Comment by amoyastudio** [comment]
Yes !!! i need to too,,,, 
and also a great great issue you can do and it's very very needed... 

in mobile - the animated text cannot be line break and i need to animate a title not one word.... it's super important... that we can have title animated and not just couple of words.... 

so we need to change the speed of the animation and also make is line break to match the div size...

hope you can make it because this will make this feature much more valuable.

**Comment by dgoooo** [comment]
+1

**Comment by adiazcastro** [comment]
+1

**Comment by black-eye** [comment]
+1

**Comment by grgentux** [comment]
It would be great since it can match the speed of a background video.

**Comment by IamJustJim** [comment]
+1

**Comment by momofone** [comment]
+1

**Comment by daankauwenberg** [comment]
+1

**Comment by nivniv44** [comment]
+1

**Comment by zhota** [comment]
+1

**Comment by aaronpeachey3** [comment]
+1

**Comment by kuznacic** [comment]
+1

**Comment by bansheekft** [comment]
speaking of the rotating style animation i actually would love to give them different speeds. the last word - or call it the final - should rest longer, then the others, to be exact: the last one needs to be shown longer, then those before for it's the gist. one customer called the default speed "childishly jittery" anotherone "frantic". my workaround (to create an animated gif) makes efforts no customer is willing to pay for. i love this feature but at the moment i don't dare showing anyone this awesome feature for it leads me to handmade animations - for every single change....

**Comment by bulutemre** [comment]
I am also struggling with this issue. I have an Animated Title on my header. It starts already before the page is loaded and you will see the start from half of it. I am not a Web developer that's why 3 days already searching for it and couldn't find anything :((

**Comment by erfan-ilyas** [comment]
+1

**Comment by mbrinson** [comment]
+1

**Comment by heyfletch** [comment]
I believe there's a need, and there's demand. :) the typing speed is too slow (3 words takes ~3 seconds, which is too long). Others are too fast, for my tastes.

**Comment by simonwidjaja** [comment]
+1
The default is too slow.

**Comment by dahianaj** [comment]
Any updates with adding the feature to control the speed of an Animated Headline? This would be really nice to have. Currently, I want to make the Rotating -> Clip version a bit slower.

**Comment by NorbertKrupa** [comment]
+1

**Comment by antonkoepke** [comment]
+1

**Comment by vicentevial** [comment]
+1

**Comment by VenkmanNL** [comment]
+1 for this feature

**Comment by lioncatde** [comment]
+1

**Comment by JimaJim** [comment]
over a year, doesn't seem like much hope but +1

**Comment by muuuhaha** [comment]
**+1**

**Comment by herooftheday77** [comment]
+1

Please guys...you can do this :-)
It's much needed and would be appreciate it.
Adjusting the speed of the animations would be a great advantage.

Or at least someone give us a timeframe...
thanks!

**Comment by LaurieCF** [comment]
+1 please :-)

**Comment by Camucamu** [comment]
+1 definitely there should be speed options

**Comment by JohJonker** [comment]
+1

**Comment by karlv-79** [comment]
Speed options are a must - please add these into the next release.

Thank you.

**Comment by amyeastment** [comment]
Piling onto this request - I'd love to be able to slow the transition down a bit for longer snippets of content.

**Comment by bongbonnggg** [comment]
+1

**Comment by aliajellu** [comment]
+1

**Comment by jackithomas** [comment]
+1

**Comment by Barns-be** [comment]
+1

**Comment by Martiniceleb** [comment]
+1

**Comment by pebblesque** [comment]
I would also love to see this feature.

**Comment by magalune** [comment]
+1

**Comment by amindiary** [comment]
This is a basic feature that must be there!
I solved my issue with custom CSS but would be good if Elementor team add this simple feature.


```
selector .elementor-headline-text-active{
     animation-duration:1s!important;
     animation-delay:0s!important
    
}
```

**Comment by jp70m** [comment]
Great CSS lines. Worked on first try!

**Comment by vicentevial** [comment]
@amindiary Hello Amin, thank you for sharing the code.

I tried adding it to the custom css part of the animated headline options with no luck (I ended up adding an extra ';' in the end, correct if should undo that please).

Here's what I've https://www.screencast.com/t/TLm0ho4DUGRB

I would really appreciate it if you can check it out and guide mi a bit, I've been waiting for this for months.

Best,
Vicente

**Comment by amindiary** [comment]
> @amindiary Hello Amin, thank you for sharing the code.
> 
> I tried adding it to the custom css part of the animated headline options with no luck (I ended up adding an extra ';' in the end, correct if should undo that please), what I'm I doing wrong?
> 
> Here's what I've https://www.screencast.com/t/TLm0ho4DUGRB
> 
> I would really appreciate it if you can check it out and guide mi a bit, I've been waiting for this for months.
> 
> Best,
> Vicente

Yes, sorry, added ; after second !important

**Comment by Codeplaya** [comment]
+1 Its too fast for my use. I need to slow down the time between animations and the above CSS does not work for me for some reason. How can there be no speed adjust on ANY animation element? I simply cannot understand that. Its obvious that for animation, the speed is one of the most important adjustments you can make. I would even go that far to say that there should be even different speed graph options, like slow on start, acceleration in the middle and and slowdown in the end again, a "fluid" checkbox representing that option. Without it, its just a not-so-wise flashing element, reminding me of web pages from Netscape era. Elementor did not sound dated like that, when it was pitching to go pro for me. So I would even be bold enough to call that a bug. It does not work like expected. And people most likely use other addons to do that job. Or just won't bother to comment in a forum, where everything is said. Or just use other page builders for the more up- to date experience.

**Comment by Kesiunas** [comment]
+1

**Comment by BWhitehair** [comment]
Created a github account just to give this a +1 :)

**Comment by amoyastudio** [comment]
i wonder why they wait so long for this simple feature.... this post has like 300 +1...

**Comment by amindiary** [comment]
> i wonder why they wait so long for this simple feature.... this post has like 300 +1...

<img src="https://i.hizliresim.com/Sz7CsD.jpg" />

**Comment by bansheekft** [comment]
what do you expect from a company that advertises a unlimited (time and pages) licence and then changes that to a "expert" one limited to one year and 1000 pages? not saying it's not worth it...
still kind of ugly - and not what i was happy to pay for...
no real response after two years...

**Comment by MB-Jan** [comment]
+1

**Comment by aliboy08** [comment]
+1

**Comment by Geekie2014** [comment]
Just popping in to say I want this!

**Comment by ytchen1** [comment]
+1 need this feature!

**Comment by GabeKeen** [comment]
This would be awesome!

**Comment by turbit-smart** [comment]
+1

**Comment by webinsightin** [comment]
> This is a basic feature that must be there!
> I solved my issue with custom CSS but would be good if Elementor team add this simple feature.
> 
> ```
> selector .elementor-headline-text-active{
>      animation-duration:1s!important;
>      animation-delay:0s!important;
>     
> }
> ```

Hi, 

This is not working for me in the latest version.
Could anyone help me?

Also, I am really curious to know why the developers cant add this basic control when clearly so many people are struggling for it?

**Comment by yorkfrog** [comment]
+1

**Comment by gonzogo** [comment]
+1 !!

**Comment by thejackk2** [comment]
https://github.com/elementor/elementor/issues/4392#issuecomment-683614545
it works well for me
You can also use milliseconds instead of seconds if needed
selector .elementor-headline-text-active{
     animation-duration:100ms!important;
     animation-delay:0s!important;
    
}

**Comment by OscarKolsrud** [comment]
This needs to happen!!!

**Comment by amoyastudio** [comment]
This needs to happen!!! you right !

**Comment by cristianuibar** [comment]
👍  +1 ... this should be in the core already.

**Comment by OzOseph** [comment]
+1.....over 2 years since this issue opened, why you didn't address that simple feature @elementor ??

**Comment by Styxwarre** [comment]
+1

**Comment by LionDes** [comment]
+1

**Comment by Kesiunas** [comment]
I'm here again with the same problem. Please add this feature asap

**Comment by webdevpro321** [comment]
December 15, 2020. Still no response huh? I guess it's time to recommend my clients that it's better to buy Divi in the future.

**Comment by webdevpro321** [comment]
Nevermind. Got it. Here's the CSS code. Tried it on the Appearance > Edit CSS. And it's working fine.

.elementor-headline--style-highlight svg path {
animation-duration: 4s !important;

/* or just edit the 10s */
animation: elementor-headline-dash 10s infinite;
}

![image](https://user-images.githubusercontent.com/76046119/102210304-f3f4db80-3f0c-11eb-82e5-971fd37dae00.png)

**Comment by amoyastudio** [comment]
@webdevpro321 Thanks

thank weird why they keep ignoring this post.... i think they don't know how much this feature is important (it's because they are the developers and not the users) , if they would build websites they also will know this one is really important

**Comment by jaysonwebtel** [comment]
+1

**Comment by jonnymueller** [comment]
+1

**Comment by neilarch** [comment]
+1

**Comment by daniswhoiam** [comment]
+1

**Comment by Ananda6262** [comment]
+1

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor Pro v3.1.0

Feel free to update

Thanks!

**Comment by larucheweb** [comment]
I feel like they don't really care about this element module... My clients love this effect but the lack of animation control is very bad. My text has a fade-in animation and by the time it's done, the text already rotated to the next line.
